### DISCUSSION TITLE
GDScript → implement named arguments

### DISCUSSION THREAD

**Discussion by azur-wolve** [godotengine/godot-proposals#4027] [Scripting]
Like C# has named arguments:
```csharp
some_func_call( arg1: value1, arg2: value2 );
```

It would allow avoiding to must include arguments in functions with several parameters.
It plays a handy role together with default args.

The syntax would be pretty much the same I guess.

**Comment by vnen** [comment]
Note that there's a conversation about this on #902, including technical challenges that goes into adding this feature to GDScript.

**Reply by azur-wolve** [reply]
Ye, I assume that the current way that GDScript internals are already crafted throws some design limitations; can't contribute too much to the discussion as don't know them.

As user-side feedback, had myself a hard time implementing a "DebugCanvas" for testing a character controller.
Tried to systematically support both fields and methods (using their "name_paths" of external entities pass through such canvas and display them, but it ended up with many parameters and hard to manage. Having named parameters could alleviate it.

But probably my design approach wasn't the appropriate. Reused an existing canvas shared by a member of the community; building over it with a different mindset become a bit messy.

**Reply by azur-wolve** [reply]
@vnen after reading yours and the others comments on #902 your position makes much more sense.

btw if eventually GDScript internals changed, lets leave it that would be a "nice-to-have++" → 𝄇
